### PR TITLE
[OpenCV 3] update to v3.4.7

### PR DESCRIPTION
**Pull request by cenit**
align the portfile to the opencv4 one to ease mainteinance

### CODE REVIEW

**Comment by vicroms** [comment]
We should have separate PRs for both versions, so that improvements on OpenCV 4 don't get delayed because of OpenCV3.

**Comment by cenit** [comment]
as of now, they are not improvements for opencv4. I am trying to uniform ports so that mainteinance can be very easy also on "older" opencv3

**Comment by cenit** [comment]
Base work should be ready.
@vicroms if you have anyone interested in opencv3, please let them try this update, then it might be ready for merging, having already all the improvements that went into opencv4.
The huge work went into the dependencies, now we are very quick in adopting new releases and if opencv3 can also receive some love then why not :)
(shall I reopen the opencv2 port PR? 🤣 )

**Comment by cenit** [comment]
I also tried to minimize *raw* diffs between opencv4 and opencv3 ports. They should look as similar as possible. Any difference should be due to something necessary, otherwise I tried to remove all of them. Naming inside the control file, names of the patches themselves, ...

File: ports/opencv3/portfile.cmake [hunk 1]
```diff
@@ -4,65 +4,66 @@ endif()
...
-set(BUILD_opencv_sfm OFF)
-if("sfm" IN_LIST FEATURES)
-  set(BUILD_opencv_sfm ON)
-endif()
-
-set(BUILD_opencv_contrib OFF)
-if("contrib" IN_LIST FEATURES)
-  set(BUILD_opencv_contrib ON)
-
+if(WITH_CONTRIB)
```
Comment (vicroms) [inline ports/opencv3/portfile.cmake]: We removed the ability for `vcpkg_check_features()` to create variables, see #7558, the `opencv4` port has been updated accordingly.

**Review by vicroms — CHANGES_REQUESTED** [review]

File: ports/opencv3/portfile.cmake [hunk 2]
```diff
@@ -4,65 +4,66 @@ endif()
...
 
 string(COMPARE EQUAL "${VCPKG_CRT_LINKAGE}" "static" BUILD_WITH_STATIC_CRT)
 
-set(CMAKE_MODULE_PATH)
-
-set(BUILD_opencv_world OFF)
-if("world" IN_LIST FEATURES)
-  set(BUILD_opencv_world ON)
-endif()
+vcpkg_check_features(
```
Comment (vicroms) [inline ports/opencv3/portfile.cmake]: missing required `OUT_FEATURE_OPTIONS` parameter, by default set to `FEATURE_OPTIONS`

**Review by vicroms — DISMISSED** [review]

**Comment by vicroms** [comment]
This LGTM, @cenit are there any other pending changes?

**Comment by vicroms** [comment]
> (shall I reopen the opencv2 port PR? 🤣 )

I know you're joking (right? 😅), but I got word from a team that might actually want to use OpenCV 2.4 😟

**Comment by cenit** [comment]
Nothing pending for now. You can merge 

I was joking, yes, but the PR for 2.4 is easy to be reopened... just let me know :)

**Comment by cenit** [comment]
there was a bug (typo) inside your last commit (FEATURE OPPTIONS with double P) https://github.com/microsoft/vcpkg/pull/7658/commits/8bd3aab803027afd5c8fb79c945abcca954807b0, which unfortunately i didn't even test so I didn't notice.
The fix is inside #7536

**Pull request merged**